lib/python/rose/checksum.py

```diff
@@ -125,21 +125,29 @@ def guess_checksum_algorithm(checksum):
 
 def _get_hexdigest(algorithm, source):
     """Load content of source into an hash object, and return its hexdigest."""
-    hashobj = hashlib.new(algorithm)
     if hasattr(source, "read"):
         handle = source
     else:
-        handle = open(source)
+        handle = open(source, 'rb')
+
+    # Attempt to find the preferred file system block size
     try:
         f_bsize = os.statvfs(handle.name).f_bsize
     except (AttributeError, OSError):
         f_bsize = 4096
+
+    # Spoon the data into a hashobj
+    hashobj = hashlib.new(algorithm)
     while True:
         bytes_ = handle.read(f_bsize)
         if not bytes_:
             break
-        hashobj.update(bytes_)
+        if type(bytes_) == bytes:
+            hashobj.update(bytes_)
+        else:
+            hashobj.update(bytes_.encode(encoding='UTF-8'))
```

Comment:
I'm slightly worried that we could end up with different hashes for the same file depending on whether the file is opened in bytes mode or not. It might be an idea to make the method use only one mode or another and update handles passed in to `_get_hexdigest`.

Reply:
I share your concern, but I've really struggled with getting this to work.

Reply:
Have raised a PR to address this: https://github.com/wxtim/rose/pull/8